roles/cloudera_deploy/tasks/auto_terraform_state.yml

```diff
@@ -0,0 +1,61 @@
+---
+
+# Copyright 2021 Cloudera, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+- name: Resources for remote_s3 state storage
+  when:
+    - globals.terraform_state_storage == 'remote_s3'
+  block:
+    
+    # Determine resource state based on auto_terraform_state_action variable
+    - name: Set variables for resource creation
+      when: auto_terraform_state_action == "create"
+      ansible.builtin.set_fact:
+        __bucket_mode: create
+        __dynamodb_table_state: present
+
+    - name: Set variables for resource teardown
+      when: auto_terraform_state_action == "teardown"
+      ansible.builtin.set_fact:
+        __bucket_mode: delete
+        __dynamodb_table_state: absent
+
+    # Create or Teardown the resources
+    - name: AWS Bucket for Remote State Storage
+      amazon.aws.aws_s3:
+        region: "{{ globals.region }}"
+        bucket: "{{ globals.terraform_remote_state_bucket}}"
+        mode: "{{ __bucket_mode }}"
```

Comment:
you could do this with a ternary on whether teardown is in the ansible tags list if you wanted to simplify it a little

Reply:
Yeah that's a nice idea. Updated the code to check on `ansible_run_tags`